r/aws_route53recoverycontrolconfig_safety_rule: panic on CreateSafetyRule error (nil pointer dereference)

### Terraform and AWS Provider Version

```console
AWS Provider v6.x (current main). The affected code is unchanged on main as of this report.
```

Originally hit through the Pulumi AWS bridge (pulumi-aws 7.39.0, which wraps this provider), but the defect is in the provider code itself and is not Pulumi specific.

### Affected Resource(s) or Data Source(s)

* `aws_route53recoverycontrolconfig_safety_rule`

### Expected Behavior

When `CreateSafetyRule` returns an error (for example a `ServiceQuotaExceededException` because a control panel already holds the maximum number of safety rules), the provider should return that error as a diagnostic.

### Actual Behavior

The provider panics with a nil pointer dereference and the process crashes before the error is ever reported.

In `internal/service/route53recoverycontrolconfig/safety_rule.go`, both `createAssertionRule` and `createGatingRule` assign `result` from the API response before checking the returned error:

```go
output, err := conn.CreateSafetyRule(ctx, input)
result := output.AssertionRule   // output is nil when err != nil

if err != nil {
    return sdkdiag.AppendErrorf(diags, "creating ... Assertion Rule: %s", err)
}
```

When `err != nil`, `output` is nil, so `output.AssertionRule` dereferences a nil pointer and panics at that line.

### Relevant Error/Panic Output

```console
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation]

github.com/hashicorp/terraform-provider-aws/internal/service/route53recoverycontrolconfig.createAssertionRule(...)
    .../internal/service/route53recoverycontrolconfig/safety_rule.go:266
github.com/hashicorp/terraform-provider-aws/internal/service/route53recoverycontrolconfig.resourceSafetyRuleCreate(...)
    .../internal/service/route53recoverycontrolconfig/safety_rule.go:130
```

The underlying API error that triggered it:

```console
api error ServiceQuotaExceededException: <control-panel-id> cannot have more than 20 SafetyRules. Current number of Safety Rules is 21
```

### Steps to Reproduce

1. Create a control panel that already has the maximum number of safety rules (or otherwise arrange for `CreateSafetyRule` to return any API error).
2. Apply a configuration that creates one more `aws_route53recoverycontrolconfig_safety_rule` on that control panel.
3. The create call returns an error, and the provider panics instead of reporting it.

### How much impact is this issue causing?

Medium

### Additional Information

Fix proposed in the linked pull request: move the `result` assignment after the error and nil-response checks in both `createAssertionRule` and `createGatingRule`.

--- a/.changelog/49155.txt
+++ b/.changelog/49155.txt
@@ -1,3 +1,3 @@
 ```release-note:bug
-resource/aws_route53recoverycontrolconfig_safety_rule: Fix panic when `CreateSafetyRule` returns an error, such as `ServiceQuotaExceededException`
+resource/aws_route53recoverycontrolconfig_safety_rule: Fix crash when the create operation returns an error
 ```

--- a/.changelog/49155.txt
+++ b/.changelog/49155.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/aws_route53recoverycontrolconfig_safety_rule: Fix panic when `CreateSafetyRule` returns an error, such as `ServiceQuotaExceededException`
+```

--- a/internal/service/route53recoverycontrolconfig/safety_rule.go
+++ b/internal/service/route53recoverycontrolconfig/safety_rule.go
@@ -272,9 +272,7 @@ func createAssertionRule(ctx context.Context, d *schema.ResourceData, meta any) 
 		return sdkdiag.AppendErrorf(diags, "creating Route53 Recovery Control Config Assertion Rule empty response")
 	}
 
-	result := output.AssertionRule
-
-	d.SetId(aws.ToString(result.SafetyRuleArn))
+	d.SetId(aws.ToString(output.AssertionRule.SafetyRuleArn))
 
 	if _, err := waitSafetyRuleCreated(ctx, conn, d.Id()); err != nil {
 		return sdkdiag.AppendErrorf(diags, "waiting for Route53 Recovery Control Config Assertion Rule (%s) to be Deployed: %s", d.Id(), err)
@@ -315,9 +313,7 @@ func createGatingRule(ctx context.Context, d *schema.ResourceData, meta any) dia
 		return sdkdiag.AppendErrorf(diags, "creating Route53 Recovery Control Config Gating Rule empty response")
 	}
 
-	result := output.GatingRule
-
-	d.SetId(aws.ToString(result.SafetyRuleArn))
+	d.SetId(aws.ToString(output.GatingRule.SafetyRuleArn))
 
 	if _, err := waitSafetyRuleCreated(ctx, conn, d.Id()); err != nil {
 		return sdkdiag.AppendErrorf(diags, "waiting for Route53 Recovery Control Config Gating Rule (%s) to be Deployed: %s", d.Id(), err)

--- a/internal/service/route53recoverycontrolconfig/safety_rule.go
+++ b/internal/service/route53recoverycontrolconfig/safety_rule.go
@@ -263,15 +263,16 @@ func createAssertionRule(ctx context.Context, d *schema.ResourceData, meta any) 
 	}
 
 	output, err := conn.CreateSafetyRule(ctx, input)
-	result := output.AssertionRule
 
 	if err != nil {
 		return sdkdiag.AppendErrorf(diags, "creating Route53 Recovery Control Config Assertion Rule: %s", err)
 	}
 
-	if result == nil {
+	if output == nil || output.AssertionRule == nil {
 		return sdkdiag.AppendErrorf(diags, "creating Route53 Recovery Control Config Assertion Rule empty response")
 	}
+
+	result := output.AssertionRule
 
 	d.SetId(aws.ToString(result.SafetyRuleArn))
 
@@ -305,15 +306,16 @@ func createGatingRule(ctx context.Context, d *schema.ResourceData, meta any) dia
 	}
 
 	output, err := conn.CreateSafetyRule(ctx, input)
-	result := output.GatingRule
 
 	if err != nil {
 		return sdkdiag.AppendErrorf(diags, "creating Route53 Recovery Control Config Gating Rule: %s", err)
 	}
 
-	if result == nil {
+	if output == nil || output.GatingRule == nil {
 		return sdkdiag.AppendErrorf(diags, "creating Route53 Recovery Control Config Gating Rule empty response")
 	}
+
+	result := output.GatingRule
 
 	d.SetId(aws.ToString(result.SafetyRuleArn))
 

